animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/ClassFileVisitor.java

```diff
@@ -123,6 +130,39 @@ else if ( ( file.getName().endsWith( ".jar" ) || file.getName().endsWith( ".jmod
         // ignore other files
     }
 
+    /**
+     * Recursively finds class files and invokes {@link #process(String, InputStream)}
+     *
+     * @param path Directory (or other Path like {@code Paths.get(URI.create("jrt:/modules"))}) full of class files,
+     *             or a class file (in which case that single class is processed).
+     */
+    public void process( Path path )
+            throws IOException {
+        Files.walkFileTree(path, Collections.<FileVisitOption>emptySet(), 10000, new SimpleFileVisitor<Path>() {
+
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.getFileName().toString().endsWith(".class")) {
+                    process(file.toString(), Files.newInputStream(file));
```

Comment:
The InputStream created by `Files.newInputStream(...)` has to be closed again because it looks like the `process(...)` method does not do that.